Signal/src/ViewControllers/MediaGalleryViewController.swift

```diff
@@ -649,8 +649,13 @@ class MediaGalleryViewController: UINavigationController, MediaGalleryDataSource
                 }
 
                 let unfetchedSet = requestSet.subtracting(self.fetchedIndexSet)
-                guard unfetchedSet.count > (requestSet.count / 2) else {
-                    // For perf we only want to fetch a relatively full batch, unless the requestSet is very small.
+
+                // For perf we only want to fetch a substantially full batch...
+                let isSubstantialRequest = unfetchedSet.count > (requestSet.count / 2)
+                // ...but we want to make sure always fulfill even small requests if we're getting just the tail end of a gallery.
+                let isFetchingEdgeOfGallery = (self.fetchedIndexSet.count - unfetchedSet.count) < requestSet.count
```

Comment:
`isFetchingEdgeOfGallery` is the only logic change, but I pulled them both out for clarity.